src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/SequenceTargetingReagentFmsDTOValidator.java

```diff
@@ -0,0 +1,90 @@
+package org.alliancegenome.curation_api.services.validation.dto.fms;
+
+import org.alliancegenome.curation_api.constants.ValidationConstants;
+import org.alliancegenome.curation_api.dao.SequenceTargetingReagentDAO;
+import org.alliancegenome.curation_api.enums.BackendBulkDataProvider;
+import org.alliancegenome.curation_api.exceptions.ObjectValidationException;
+import org.alliancegenome.curation_api.model.entities.SequenceTargetingReagent;
+import org.alliancegenome.curation_api.model.entities.ontology.NCBITaxonTerm;
+import org.alliancegenome.curation_api.model.ingest.dto.fms.SequenceTargetingReagentFmsDTO;
+import org.alliancegenome.curation_api.response.ObjectResponse;
+import org.alliancegenome.curation_api.response.SearchResponse;
+import org.alliancegenome.curation_api.services.DataProviderService;
+import org.alliancegenome.curation_api.services.ontology.NcbiTaxonTermService;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.inject.Inject;
+
+@RequestScoped
+public class SequenceTargetingReagentFmsDTOValidator {
+	@Inject
+	DataProviderService dataProviderService;
+
+	@Inject
+	SequenceTargetingReagentDAO sqtrDAO;
+
+	@Inject NcbiTaxonTermService ncbiTaxonTermService;
+
+	public SequenceTargetingReagent validateSQTRFmsDTO(SequenceTargetingReagentFmsDTO dto, BackendBulkDataProvider beDataProvider) throws ObjectValidationException {
+		ObjectResponse<SequenceTargetingReagent> sqtrResponse = new ObjectResponse<>();
+		
+		SequenceTargetingReagent sqtr;
+		
+		if (StringUtils.isBlank(dto.getPrimaryId())) {
+			sqtrResponse.addErrorMessage("primaryId", ValidationConstants.REQUIRED_MESSAGE);
+			sqtr = new SequenceTargetingReagent();
+		} else {
+			SearchResponse<SequenceTargetingReagent> searchResponse = sqtrDAO.findByField("curie", dto.getPrimaryId());
```

Comment:
`curie` is reserved for AGR curies for `SubmittedObject` descendants, so lookup and storing of `primaryId` should be with `modEntityId`

Reply:
Cool, I will make that change.

Reply:
Good catch